src/utils/thread_helper.cpp

```diff
@@ -920,6 +920,42 @@ void ThreadHelper::DetachGracefullyCallback(void)
 }
 
 #if OTBR_ENABLE_TELEMETRY_DATA_API
+#if OTBR_ENABLE_BORDER_ROUTING
+void ThreadHelper::GetNodeDivergenceInfo(threadnetwork::TelemetryData_NodeDivergenceInfo *aNodeDivergenceInfo)
+{
+    bool isExternalRoutePublished = false;
+    bool isDefaultRoutePublished  = false;
+    Ip6Prefix prefix;
+    uint16_t rloc16               = otThreadGetRloc16(mInstance);
+
+    otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
+    otExternalRouteConfig config;
+
+    // iterate over leader network data
+    while (otNetDataGetNextRoute(mInstance, &iterator, &config) == OT_ERROR_NONE)
+    {
+        // filter out the config that is not stable or unmatched rloc16
+        if (!config.mStable || config.mRloc16 != rloc16)
+        {
+            continue;
+        }
+
+        // network data has the BR's external route
+        isExternalRoutePublished = true;
+
+        // network data has the default route of the BR if the prefix is matched
+        prefix.Set(config.mPrefix);
+        if (prefix.IsDefaultRoutePrefix())
+        {
+            isDefaultRoutePublished  = true;
```

Comment:
Do you want to break out the loop when this value is set to true?

Reply:
I would like to iterate over in case there are multiple entries

Reply:
When isDefaultRoutePublished = true, it seems that this is not to be assigned back to false again, do you need any more iterations?

Reply:
good point, update! will break after isDefaultRoutePublished set to true.